pkg/types/validation/installconfig.go

```diff
@@ -126,40 +126,54 @@ func ValidateInstallConfig(c *types.InstallConfig) field.ErrorList {
 	return allErrs
 }
 
-// ipAddressTypeByField is a map of field path to whether they request IPv4 or IPv6.
-type ipAddressTypeByField map[string]struct{ IPv4, IPv6 bool }
+// ipAddressType indicates the address types provided for a given field
+type ipAddressType struct {
+	IPv4        bool
+	IPv4Primary bool
```

Comment:
can it be IPv4Primary and IPv6Primary both true?

Reply:
no... I guess I could do `Primary v1.IPFamily` or something instead?